src/compiler/checker.ts

```diff
@@ -12798,6 +12804,14 @@ namespace ts {
             return undefined;
         }
 
+        function getParameterTypeFromBody(parameter: ParameterDeclaration): Type {
+            const func = <FunctionLikeDeclaration>parameter.parent;
+            if (!func.body || isRestParameter(parameter)) return;
+
+            const types = checkAndAggregateParameterExpressionTypes(parameter);
+            return types ? getWidenedType(getIntersectionType(types)) : undefined;

```

Comment:
@HerringtonDarkholme Could you provide a concrete example of this? For which reference is `someCond` affecting the type?